json-to-messages/javascript/src/JSONTraverse.ts

```diff
@@ -8,15 +8,16 @@ import { traverseFeature as traverseBehaveFeature } from './behave/JSONTraverse'
 import IPredictableSupportCode from './IPredictableSupportCode'
 import IAstMaker from './IAstMaker'
 import { IdGenerator, messages } from '@cucumber/messages'
+import { Platform } from './types'
 
 export default function traverseFeature(
-  sourceLang: string,
+  platform: Platform,
   feature: IFeature | IBehaveFeature,
   astMaker: IAstMaker,
   newId: IdGenerator.NewId,
   predictableSupportCode: IPredictableSupportCode
 ): messages.IGherkinDocument {
-  if (sourceLang === 'javascript') {
+  if (platform === 'javascript') {
```

Comment:
I could use the java impl to generate messages that would be classed as JavaScript. 

Perhaps gherkin and behave would be better.

Reply:
Well, it' not as simple a Gherkin vs Behave (Behave uses Gherkin, not the library but at leat the language).

Here we mainly detect based on specificities of the existing reports. If fact, we don't even have a `cucumber-java` platform/language/dialect as the JSON reports are close enough to `cucumber-ruby`, so we use the same object to traverse.